tools/data_integrity/score_instrument.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * This tools scores any registered instrument that was build using the
+ * NDB_BVL_Instrument class and that has a working score() method.
+ * The cmd-line arguments need to contain a valid test_name, 'all' or 'one' and a
+ * CandID and SessionID. 'all' scores all records of an instrument that belong to an
+ * active timepoint and whose data entry is completed and `administration` not equal to  'None'.
+ * The non-administered records' scores have to be reset in the DB.
+ *
+ * This tool can also be used to debug the scoring:
+ * // To debug, comment out the UPDATE to DB in the scoring method(s) of the instrument's class
+ * // add return $scores - an array of scores you're sending to the update
+ * // and uncomment the lines at the end of the script to print out scores on the screen.
+ *
+ * @version $Id: score_instrument.php,v 1.2 2006/04/07 12:58:41 jharlap Exp $
+ * @package behavioural
+ */
+
+require_once __DIR__."/../generic_includes.php";
+
+// LOGGING
+$dir = __DIR__ . "/../logs/";
+if (!is_dir($dir)) {
+    mkdir($dir);
+}
+$today= getdate();
+$date = strftime("%Y-%m-%d_%H:%M");
+$logPath = "$dir/score_instrument.$date.log";
+$logfp = fopen($logPath, 'a');
+
+if (!$logfp) {
+    printError(
+        "No logs can be generated, path:$logPath ".
+        "does not exist or can not be written to.\n"
+    );
+}
+
+/**
+ * HELP SCREEN
+ * display and stop processing if action=help
+ */
+if (empty($argv[1])
+    || $argv[1] == 'help'
```

Comment:
might be good to have an argument as well for `-h`. My first reflex was to type `php score_instruments.php -h`